py/torch_tensorrt/fx/converters/operator.py

```diff
@@ -0,0 +1,1256 @@
+    input_val = kwargs["input"]
+
+    if not isinstance(input_val, TRTTensor):
+        raise RuntimeError(
+            f"Linear received input {input_val} that is not part "
+            "of the TensorRT region!"
+        )
+
+    dynamic_dims = get_dynamic_dims(input_val.shape)
+    assert len(dynamic_dims) < 2 and input_val.shape[-1] != -1, (
+        "Currently we only support one dynmaic "
+        "dim for linear and it can't be the last dim."
+    )
+
+    if isinstance(kwargs["weight"], torch.Tensor):
+        weight = get_trt_tensor(network, kwargs["weight"].t(), f"{name}_weight")
+        if target not in (acc_ops.linear, torch.ops.aten.linear):
+            weight_op = trt.MatrixOperation.TRANSPOSE
+        else:
+            weight_op = trt.MatrixOperation.NONE
+    else:
+        assert isinstance(
+            kwargs["weight"], TRTTensor
+        ), f"Expect weight to be trt tensor but got {type(kwargs['weight'])}"
+        weight = kwargs["weight"]
+        weight_op = trt.MatrixOperation.TRANSPOSE
+
+    preset_diff = 0
+    if len(input_val.shape) == 1:
+        preset_diff -= 1
+        input_op = trt.MatrixOperation.VECTOR
+    else:
+        input_op = trt.MatrixOperation.NONE
+
+    input_val, weight = broadcast(
+        network, input_val, weight, f"{name}_input", f"{name}_weight", preset_diff
+    )
+    matmul_layer = network.add_matrix_multiply(input_val, input_op, weight, weight_op)
+    set_layer_name(matmul_layer, target, f"{name}_matmul")
+    res = matmul_layer.get_output(0)
+
+    if kwargs["bias"] is not None:
+        bias = get_trt_tensor(network, kwargs["bias"], f"{name}_bias")  # type: ignore[arg-type]
+        res = add_binary_elementwise_layer(
+            network,
+            matmul_layer.get_output(0),
+            bias,
+            trt.ElementWiseOperation.SUM,
+            target,
+            f"{name}_add",
+        )
+    return res
+
+
+def add_unary_layer(
+    network: TRTNetwork,
+    input_val: TRTTensor,
+    operation_type: trt.UnaryOperation,
+    target: Target,
+    name: str,
+) -> TRTTensor:
+    """
+    Add a TensorRT Unary layer to `network`.
+
+    Args:
+        network (TRTNetwork): TensorRT network object.
+        input_val (TRTTensor): Input to the unary op. Must be a TensorRT tensor.
+        op_type (trt.ElementWiseOperation): Type of the TensorRT unary operation.
+        target (Target): Target of fx node.
+        name (str): The name we want to assign to the created TensorRT layer.
+
+    Returns:
+        The output of TensorRT Unary layer.
+    """
+    if not isinstance(input_val, TRTTensor):
+        raise RuntimeError(
+            f"{operation_type} received input {input_val} that is not part "
+            "of the TensorRT region!"
+        )
+    layer = network.add_unary(input_val, operation_type)
+    set_layer_name(layer, target, name)
+    output = layer.get_output(0)
+    output.name = output.name + "_" + target.__name__
+    return layer.get_output(0)
```

Comment:
Does `layer.get_output(0)` make a copy of the underlying object? If so, line 547 may have no effect, since it only modifies the `output` object. Should this be `return output` instead?

Reply:
So the `layer.get_output(0) `does not make an explicit copy and modifying the `output.name` causes inplace modification of the `layer.get_output(0).name` too. Yes it could have been `return output` too I assume, but any change would be inconsequential.